### PR TITLE
BugFix: `compute()` explicitly rejects a promise if buffer transferring fails

### LINKED ISSUE #634: `MLContext.compute()` explicitly rejects promises where the sub-septs may currently throw
Regarding to the existing [`MLContext.compute()` algorithm](https://www.w3.org/TR/webnn/#dom-mlcontext-compute), if input validation steps fail, it returns a rejected promise. However, in the following two steps, [transferring an MLNamedArrayBufferViews steps](https://www.w3.org/TR/webnn/#mlnamedarraybufferviews-transfer) may throw an exception because [transfer an ArrayBuffer steps](https://webidl.spec.whatwg.org/#arraybuffer-transfer) will throw a TypeError if [DetachArrayBuffer](https://tc39.es/ecma262/multipage/structured-data.html#sec-detacharraybuffer) fails. 

> 7. Let transferredInputs be the result of transferring MLNamedArrayBufferViews inputs with realm.

> 8. Let transferredOutputs be the result of transferring MLNamedArrayBufferViews outputs with realm.

Although the exceptions thrown by step 7 and 8 should be converted to promise rejection automatically, according https://www.w3.org/2001/tag/doc/promises-guide#errors - that says

> For Web IDL-based specs, this is taken care of automatically if you declare your [operations](https://heycam.github.io/webidl/#dfn-operation) to return a [promise type](https://heycam.github.io/webidl/#dfn-promise-type). Any exceptions thrown by such operations, or by the Web IDL-level type conversions and overload resolution, are automatically converted into rejections.

@inexorabletash suggested

> being explicit for subtle things like this is a good idea. Something like "If that throws an exception, return a promise rejected with the exception".

/cc @a-sully

### PR DESCRIPTION
**Pull request by huningxin**
Fix #634

/cc @a-sully

***
<a href="https://pr-preview.s3.amazonaws.com/huningxin/webnn/pull/639.html" title="Last updated on Apr 8, 2024, 6:01 AM UTC (a2560c3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/639/2f32429...huningxin:a2560c3.html" title="Last updated on Apr 8, 2024, 6:01 AM UTC (a2560c3)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -919,8 +919,8 @@ In accordance with the [=ArrayBufferView/write|Web IDL warning=], to prevent the
     1. If |graph|.{{MLGraph/[[context]]}}.{{MLContext/[[contextType]]}} is not "[=context type/default=]", then return [=a new promise=] [=rejected=] with an "{{OperationError}}" {{DOMException}}.
     1. If [=validating graph resources=] given |inputs| and |graph|.{{MLGraph/[[inputDescriptors]]}} returns false, then return [=a new promise=] [=rejected=] with a {{TypeError}}.
     1. If [=validating graph resources=] given |outputs| and |graph|.{{MLGraph/[[outputDescriptors]]}} returns false, then return [=a new promise=] [=rejected=] with a {{TypeError}}.
-    1. Let |transferredInputs| be the result of [=MLNamedArrayBufferViews/transfer|transferring=] {{MLNamedArrayBufferViews}} |inputs| with |realm|.
-    1. Let |transferredOutputs| be the result of [=MLNamedArrayBufferViews/transfer|transferring=] {{MLNamedArrayBufferViews}} |outputs| with |realm|.
+    1. Let |transferredInputs| be the result of [=MLNamedArrayBufferViews/transfer|transferring=] {{MLNamedArrayBufferViews}} |inputs| with |realm|. If that threw an exception, return [=a new promise=] [=rejected=] with that exception.
```
Comment (inexorabletash) [inline index.bs:922]:
```suggestion
    1. Let |transferredInputs| be the result of [=MLNamedArrayBufferViews/transfer|transferring=] {{MLNamedArrayBufferViews}} |inputs| with |realm|. If that threw an exception, then return [=a new promise=] [=rejected=] with that exception.
```

File: index.bs [hunk 2]
```diff
@@ -919,8 +919,8 @@ In accordance with the [=ArrayBufferView/write|Web IDL warning=], to prevent the
...
+    1. Let |transferredOutputs| be the result of [=MLNamedArrayBufferViews/transfer|transferring=] {{MLNamedArrayBufferViews}} |outputs| with |realm|. If that threw an exception, return [=a new promise=] [=rejected=] with that exception.
```
Comment (inexorabletash) [inline index.bs:923]:
```suggestion
    1. Let |transferredOutputs| be the result of [=MLNamedArrayBufferViews/transfer|transferring=] {{MLNamedArrayBufferViews}} |outputs| with |realm|. If that threw an exception, then return [=a new promise=] [=rejected=] with that exception.
```

**Review by inexorabletash — APPROVED** [review]
LGTM with minor suggestions.

File: index.bs (hunk 1 above)
Comment (huningxin) [reply index.bs]: Fixed in https://github.com/webmachinelearning/webnn/pull/639/commits/a2560c3acf37327df0eefbeb2ae5908b0e11bba1

File: index.bs (hunk 2 above)
Comment (huningxin) [reply index.bs]: Fixed in https://github.com/webmachinelearning/webnn/pull/639/commits/a2560c3acf37327df0eefbeb2ae5908b0e11bba1.

**Comment by huningxin** [comment]
@fdwr , any further comments?

**Review by fdwr — APPROVED** [review]

**Pull request merged**